assertion-analyser.js

```diff
@@ -60,12 +62,13 @@ function objParser(str, init) {
 
 function replacer(str) {
   // replace objects with a symbol ( __#n)
-  var obj;
-  var cnt = 0;
-  var data = [];
-  while(obj = objParser(str)) {
+  let cnt = 0;
+  let data = [];
+  let obj = objParser(str);
+  while(obj) {
     data[cnt] = obj.obj;
-    str = str.substring(0, obj.start) + '__#' + cnt++ + str.substring(obj.end+1)
+    str = str.substring(0, obj.start) + '__#' + cnt++ + str.substring(obj.end+1);
+    objParser(str);
```

Comment:
```suggestion
    obj = objParser(str);
```

After a call with @RandellDawson, we discovered that the test calls here will not run because the `obj` variable is not reassigned.

Reply:
@raisedadead @SaintPeter Can you commit this change?  It fixes a bug. We are unable to push any commits to this branch.

Reply:
Only Rex is able to make changes to this branch. Alternatively, we can land this on a temp branch and make further changes. Let me know.